src/annotator/anchoring/text-range.js

```diff
@@ -128,15 +131,43 @@ export class TextPosition {
   /**
    * Resolve the position to a specific text node and offset within that node.
    *
-   * Throws if `this.offset` exceeds the length of the element's text or if
-   * the element has no text. Offsets at the boundary between two nodes are
-   * resolved to the start of the node that begins at the boundary.
+   * Throws if `this.offset` exceeds the length of the element's text. In the
+   * case where the element has no text and `this.offset` is 0, the `dir` option
+   * determines what happens.
+   *
+   * Offsets at the boundary between two nodes are resolved to the start of the
+   * node that begins at the boundary.
    *
+   * @param {Object} [options]
+   *   @param {RESOLVE_FORWARDS|RESOLVE_BACKWARDS} [options.dir] -
+   *     Specifies in which direction to search for the nearest text node if
+   *     `this.offset` is `0` and `this.element` has no text. If not specified
+   *     an error is thrown.
```

Comment:
I don't love the `dir` option name but the meaning is tricky to capture in an obvious name.